generators/import-jdl/index.js

```diff
@@ -154,93 +129,85 @@ module.exports = class extends BaseGenerator {
-                                });
-                            } else {
-                                generateEntityFiles(this, entity);
-                            }
-                        });
+                    this.importState.exportedEntities.forEach((exportedEntity) => {
+                        if (this.importState.exportedApplications.length === 0
+                                || this.importState.exportedApplications.length === 1) {
+                            this.log(`Generating ${this.importState.exportedEntities.length} entities.`);
+                            generateEntityFiles(this, exportedEntity);
+                        } else {
+                            // sub-folder generation, not yet handled
+                            this.entitiesLeftToGenerate.push(exportedEntity.name);
+                        }
+                    });
                 } catch (error) {
-                    this.error(`While generating applications and entities from the parsed JDL\n${error}`);
+                    this.error(`While generating entities from the parsed JDL\n${error}`);
                 }
             }
         };
     }
 
     end() {
         if (!this.options['skip-install'] && !this.skipClient && !this.options['json-only']
-                && !shouldGenerateApplications(this, this.jdlObject)) {
+            && !shouldGenerateApplications(this, this.jdlObject)) {
             this.debug('Building client');
             this.rebuildClient();
         }
+        if (this.applicationsLeftToGenerate.length !== 0) {
+            this.info(`Here are the application names to generate: ${this.applicationsLeftToGenerate.join(', ')}`);
+        }
+        if (this.entitiesLeftToGenerate.length !== 0) {
+            this.info(`Here are the entity names to generate: ${_.uniq(this.entitiesLeftToGenerate).join(', ')}`);
+        }
     }
 };
 
-function shouldGenerateApplications(generationConfiguration, jdlObject) {
-    return !generationConfiguration.options['ignore-application'] && Object.keys(jdlObject.applications).length !== 0;
-}
-
-function getApplicationPaths(jdlApplications) {
-    const paths = {};
-    Object.keys(jdlApplications).forEach((baseName) => {
-        paths[baseName] = jdlApplications[baseName].config.path;
-    });
-    return paths;
-}
-
-function setGeneratorVersionInJDLApplications(jdlApplications, version) {
-    Object.keys(jdlApplications).forEach((applicationName) => {
-        jdlApplications[applicationName].config.jhipsterVersion = version;
-    });
-    return jdlApplications;
+function shouldGenerateApplications(generator) {
+    return !generator.options['ignore-application'] && generator.importState.exportedApplications.length !== 0;
 }
 
 function generateApplicationFiles(generator, application) {
-    const initialWorkingDir = shelljs.pwd();
-    shelljs.cd(path.join(shelljs.pwd().stdout, application.path || ''));
     callSubGenerator(generator, '..', 'app', {
```

Comment:
Logically, that's how a subgen should be called (if I'm not wrong). And after looking at the code for the app subgen, I can't understand why it doesn't pick up the .yo-rc.json file. The creation of the files takes place before the subgens are called (if I understand Yeoman precedence rules).